### PR TITLE
July 2023

### PR DESCRIPTION
**Pull request by SethBarberee**
Quite a lot of work combining and splitting files. It seems like I found a few more files that deal with kanghaskhan and gulpin.

- Cleaned up a few headers and other files that included the relevant functions.
- Added more files to ewram generation

### CODE REVIEW

File: src/code_801EE10_mid.c [hunk 1]
```diff
@@ -224,3 +254,178 @@ u32 sub_801F7E4(void)
...
+    gUnknown_203B278->moveIDs = moveIDs;
+
+    for(index = 4; index > 1; index--)
+    {
+        if(gUnknown_203B278->moveIDs[index - 1] != 0) break;
+    }
+    sub_8013818(&gUnknown_203B278->unk28,index,1,gUnknown_203B278->unk5C);
+    sub_801317C(&gUnknown_203B278->unkC8);
+    sub_801F918(0);
+    return 1;
```
Comment (AnonymousRandomPerson) [inline src/code_801EE10_mid.c:278]: `return TRUE`

File: src/exclusive_pokemon.c [hunk 2]
```diff
@@ -49,3 +48,269 @@ void InitializeExclusivePokemon(void)
...
+    s32 pokeID_s32_1;
+    struct ExclusivePokemonData *ptr;
+
+    pokeID_s32 = pokeID;
+    pokeID_s32_1 = pokeID_s32;
+    if (pokeID_s32 != MONSTER_DECOY)
+    {
+        if(pokeID_s32 != MONSTER_STATUE)
+        {
+            if(pokeID_s32 != MONSTER_RAYQUAZA_CUTSCENE) {
```
Comment (AnonymousRandomPerson) [inline src/exclusive_pokemon.c:121]: Is it possible to combine these three nested ifs?

File: src/kecleon_items_1.c [hunk 3]
```diff
@@ -11,11 +11,26 @@
 #include "menu_input.h"
 #include "text.h"
 
+struct unkStruct_203B214
+{;
```
Comment (AnonymousRandomPerson) [inline src/kecleon_items_1.c:15]: Extra semicolon.

File: src/exclusive_pokemon.c (hunk 2 above)
Comment (SethBarberee) [reply src/exclusive_pokemon.c:121]: Looks like it tries to optimize it to a subtraction if instead of each case:
https://decomp.me/scratch/zuPpI

Comment (AnonymousRandomPerson) [reply src/exclusive_pokemon.c:121]: Ah, nvm then.

**Review by AnonymousRandomPerson — APPROVED** [review]

File: src/gulpin_shop_1.c [hunk 4]
```diff
@@ -702,15 +699,13 @@ void sub_8020950(void)
     gUnknown_203B27C->unk7C[loopMax].menuAction = 1;
 }
 
-extern s32 sub_80144A4(s32 *);
-
 void sub_80209AC(void)
 {
-    s32 temp;
-    if(sub_80144A4(&temp) == 0)
+    s32 menuAciton;
```
Comment (Kermalis) [inline src/gulpin_shop_1.c:704]: menuAciton

**Pull request merged**